packages/opentelemetry-plugin-xml-http-request/test/xhr.test.ts

```diff
@@ -478,144 +749,416 @@ describe('xhr', () => {
           webTracerWithZone = webTracerWithZoneProvider.getTracer('xhr-test');
 
           rootSpan = webTracerWithZone.startSpan('root');
+        });
 
-          webTracerWithZone.withSpan(rootSpan, () => {
-            getData(
-              url,
-              () => {
-                fakeNow = 100;
-              },
-              testAsync
-            ).then(() => {
-              fakeNow = 0;
-              sandbox.clock.tick(1000);
-              done();
+        afterEach(() => {
+          clearData();
+        });
+
+        describe('when request loads and receives an error code', () => {
+          beforeEach(done => {
+            webTracerWithZone.withSpan(rootSpan, () => {
+              getData(
+                url,
+                () => {
+                  fakeNow = 100;
+                },
+                testAsync
+              ).then(() => {
+                fakeNow = 0;
+                sandbox.clock.tick(1000);
+                done();
+              });
+              assert.strictEqual(requests.length, 1, 'request not called');
+              requests[0].respond(
+                400,
+                { 'Content-Type': 'text/plain' },
+                'Bad Request'
+              );
             });
-            assert.strictEqual(requests.length, 1, 'request not called');
-            requests[0].respond(
+          });
+          it('span should have correct attributes', () => {
+            const span: tracing.ReadableSpan = exportSpy.args[0][0][0];
+            const attributes = span.attributes;
+            const keys = Object.keys(attributes);
+
+            assert.ok(
```

Comment:
I see that you added various request unsuccessful cases! The test cases look good.

I noticed that the test file became a bit lengthy and am wondering if we could maybe factor out the span attribute testing into a function and call that function instead (I am unsure if this would be a good idea since we using too many helper functions is usually discouraged in test files for easier readability, so I may ask for Eryn's opinion as well).